src/Utilities/Extensions/ISymbolExtensions.cs

```diff
@@ -268,9 +268,19 @@ public static bool ParameterTypesAreSame(this IMethodSymbol method1, IMethodSymb
             {
                 cancellationToken.ThrowIfCancellationRequested();
 
+                var type1 = method2.Parameters[index].Type.OriginalDefinition;
+                var type2 = method1.Parameters[index].Type.OriginalDefinition;
+
+                if (type1.TypeKind == TypeKind.TypeParameter &&
```

Comment:
This matches our comparison in Roslyn: http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Shared/Utilities/SymbolEquivalenceComparer.EquivalenceVisitor.cs,570. Ideally, we would be able to use Roslyn's [SymbolEquivalenceComparer](http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Shared/Utilities/SymbolEquivalenceComparer.cs,f93cc1c775e51f8b), but that is internal.